crates/red_knot/src/types/infer.rs

```diff
@@ -70,8 +70,27 @@ where
                 let ast = parsed.ast();
                 let node = node_key.resolve_unwrap(ast.as_any_node_ref());
 
+                let bases: Vec<_> = node
+                    .bases()
+                    .iter()
+                    .map(|base_expr| match base_expr {
+                        ast::Expr::Name(name) => {
+                            if let Some(base_symbol_id) = symbols.root_symbol_id_by_name(&name.id) {
+                                db.jar().type_store.record_symbol_dependency(
```

Comment:
It would be nice if `root_symbol_id_by_name` to have an automated way to record the dependencies between symbols, especially if `symbols.root_symbol_id_by_name` is a public method (that can also be used by e.g. semantic lint rules!).

Reply:
Yes, totally agree! I thought I already added a TODO for that somewhere in here? I might not do it in this PR because I'd like to consider the API a bit carefully, but I definitely want to do it soon.

Reply:
Oh I added the TODO in https://github.com/astral-sh/ruff/pull/11176 which this PR is based on.

Reply:
This will be part of the dependency-recording PR, it's gone from this PR now.